python/lsst/ts/mtdome/power_management/power_management_handler.py

```diff
@@ -0,0 +1,486 @@
+
+        This method takes the maintenance mode priorities into account. The
+        priorities, from high to low, are:
+
+          * Move/crawl the light/wind screen.
+          * Enable the fans.
+
+        If the light/wind screen is scheduled to be moved and the fans are
+        currently on, the fans first are stopped. Only then the light/wind
+        screen will be moved.
+        If the fans are scheduled to be switched on and the light/wind screen
+        currently is moving, then switching on the fans will wait until the
+        light/wind screen motion is done.
+        To be on the safe side, if the Aperture Shutter or Louvers are moving,
+        that motion first will be stopped.
+        Any other command will be refused.
+
+        Parameters
+        ----------
+        current_power_draw : `dict`[`LlcName`, `float`]
+            Dict of the current power draw [W] for the subsystems, or Lower
+            Level Components.
+
+        Returns
+        -------
+        ScheduledCommand | None
+            The next command to be issued, or None if no commands currently can
+            be issued or are scheduled.
+
+        Notes
+        -----
+        It is assumed that the Rear Access Door, the Overhead Bridge Crane and
+        the Calibration screen may, and the Aperture Shutter and Louvers may
+        not be used in this mode. The latter is true so the environment inside
+        the dome can be controlled.
+        """
+        _, scheduled_command = await self.command_queue.get()
+        match scheduled_command.command:
+            case CommandName.MOVE_EL | CommandName.CRAWL_EL:
+                return await self.issue_scheduled_command_or_stop_commands(
+                    scheduled_command,
+                    current_power_draw,
+                    [STOP_SHUTTER, STOP_LOUVERS, STOP_FANS],
+                    [],
+                )
+            case CommandName.FANS:
+                # Switching off the fans is always allowed.
+                if scheduled_command.params["action"] == OnOff.OFF:
+                    return scheduled_command
+                else:
+                    return await self.issue_scheduled_command_or_stop_commands(
+                        scheduled_command,
+                        current_power_draw,
+                        [STOP_SHUTTER, STOP_LOUVERS],
+                        [LlcName.LWSCS],
+                    )
+            case (
+                CommandName.OPEN_SHUTTER
+                | CommandName.CLOSE_SHUTTER
+                | CommandName.SEARCH_ZERO_SHUTTER
+                | CommandName.SET_LOUVERS
+                | CommandName.CLOSE_LOUVERS
+            ):
+                # These commands are not allowed and therefore are ignored. By
+                # doing so, they automatically are removed from the queue.
+                return None
+            case _:
+                return scheduled_command
+
+    async def get_next_command_emergency(
```

Comment:
See my comment above about using "get_<something>" to name this method.

Reply:
See my reply above.